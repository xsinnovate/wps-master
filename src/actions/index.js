export const PROMO_CODE = 'PROMO_CODE';

export const applyPromoCode = code => ({ type: PROMO_CODE, code });
