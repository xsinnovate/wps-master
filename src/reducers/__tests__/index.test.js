import { applyPromoCode } from '../../actions';
import reducer, { initialState, getCartProducts, getCartTotals } from '..';

/**
 * NOTE: Snapshot is a shortcut. One could also be more explicit, e.g.
 *
 * expect(reducer(initialState, applyPromoCode('DISCOUNT'))).toEql({
 *   ...initialState,
 *   promo: 10,
 * });
 */

describe('rootReducer', () => {
  it('should have initial cart state', () => {
    expect(reducer(initialState, {})).toMatchSnapshot();
  });

  it('should add 10% discount if promo code DISCOUNT used', () => {
    expect(reducer(initialState, applyPromoCode('DISCOUNT'))).toMatchSnapshot();
  });

  it('should not apply discount for other promo codes', () => {
    expect(reducer(initialState, applyPromoCode('INVALID'))).toMatchSnapshot();
  });
});

describe('getCartProducts()', () => {
  it('should return the array of products in cart with details', () => {
    expect(getCartProducts(initialState)).toMatchSnapshot();
  });
});

describe('getCartTotals()', () => {
  it('should return an object with the calculated cart totals', () => {
    expect(getCartTotals(initialState)).toMatchSnapshot();
  });
});
