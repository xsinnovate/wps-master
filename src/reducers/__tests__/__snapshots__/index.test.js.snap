// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCartProducts() should return the array of products in cart with details 1`] = `
Array [
  Object {
    "id": "prod1",
    "image": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
    "name": "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
    "pickup": 3.85,
    "price": 102.96,
    "quantity": 1,
  },
]
`;

exports[`getCartTotals() should return an object with the calculated cart totals 1`] = `
Object {
  "promo": 0,
  "savings": 3.85,
  "subtotal": 102.96,
  "taxes": 8.92,
  "total": 108.03,
}
`;

exports[`rootReducer should add 10% discount if promo code DISCOUNT used 1`] = `
Object {
  "cart": Array [
    Object {
      "id": "prod1",
      "quantity": 1,
    },
  ],
  "products": Object {
    "prod1": Object {
      "image": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
      "name": "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
      "pickup": 3.85,
      "price": 102.96,
    },
  },
  "promo": 10,
}
`;

exports[`rootReducer should have initial cart state 1`] = `
Object {
  "cart": Array [
    Object {
      "id": "prod1",
      "quantity": 1,
    },
  ],
  "products": Object {
    "prod1": Object {
      "image": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
      "name": "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
      "pickup": 3.85,
      "price": 102.96,
    },
  },
  "promo": 0,
}
`;

exports[`rootReducer should not apply discount for other promo codes 1`] = `
Object {
  "cart": Array [
    Object {
      "id": "prod1",
      "quantity": 1,
    },
  ],
  "products": Object {
    "prod1": Object {
      "image": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
      "name": "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
      "pickup": 3.85,
      "price": 102.96,
    },
  },
  "promo": 0,
}
`;
